core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1455,7 +1456,11 @@ private[spark] object Utils extends Logging {
     val effectiveEnd = math.min(length, end)
     val effectiveStart = math.max(0, start)
     val buff = new Array[Byte]((effectiveEnd-effectiveStart).toInt)
-    val stream = new FileInputStream(file)
+    val stream = if (path.endsWith(".gz")) {
```

Comment:
oh, Utils.scala doesn't depend on `FileAppender`. It does use string literal in other places. https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/Utils.scala#L463